lib/ex_doc/doc_ast.ex

```diff
@@ -5,9 +5,9 @@ defmodule ExDoc.DocAST do
   alias ExDoc.Formatter.HTML.Templates
 
   @doc """
-  Parse given `doc` according to `doc_format`.
+  Parses given `doc_content_binary` according to `doc_format`.
   """
-  def parse!(doc, doc_format, options \\ [])
+  def parse!(doc_content_binary, doc_format, options \\ [])
```

Comment:
it's not a binary for `"application/erlang+html"` :)

Reply:
should we name it `doc_content_binary_or_ast` ?

Reply:
how about just doc_content?

Reply:
> how about just doc_content?

because doc_content according  the specs is the map that contains this string for Elixir docs.

`doc_content: %{optional(binary()) => binary()} | :none | :hidden,`

Reply:
we can just leave `doc` and explain in the docs what `doc` could be, either an erlang ast or a string containing markdown

Reply:
Something like this.

```elixir
  @doc """
  Parses given `doc` according to `doc_format`.

  When `doc_format` is "text/markdown", `doc` is expeceted to be a binary.
  When `doc_format` is "application/erlang+html", `doc` is expeceted to be an Erlang AST.
  """
  def parse!(doc, doc_format, options \\ [])
```

Reply:
A friendly reminder that this is a private module, we don’t need to add comprehensive docs

Reply:
Yes, I know but I am working on a fix and everthing is named "doc" and "docs" and  it is hard to grasp what is what.

Reply:
I am reverting this change